msteams-platform/messaging-extensions/build-bot-based-plugin.md

```diff
@@ -0,0 +1,242 @@
+
+1. From the compose message area, send a message to Copilot to find an NuGet package information. For example: Find the NuGet package info on Microsoft.CSharp.
+
+   :::image type="content" source="../assets/images/Copilot/bot-based-VSC-trigger-plugin.png" alt-text="Screenshot shows the plugin prompt and the response from copilot.":::
+
+> [!NOTE]
+> This prompt may not always make Copilot include a response from your message extension. If it happens, try some other prompts or leave feedback to us by thumbing down the Copilot response and leave a message tagged with [MessageExtension].
+
+# [Visual Studio](#tab/visual-studio)
+
+1. Open Visual Studio.
+1. Go to **File** > **New** > **Project...** or **New Project**.
+
+1. Search for **Teams** and select **Microsoft Teams App**.
+
+1. Select **Custom Search Results**.
+
+   :::image type="content" source="../assets/images/Copilot/bot-based-VS-new-project.png" alt-text="Screenshot shows the custom search results option in Visual Studio.":::
+
+1. Select **Create**. The project is scaffolded.
+
+1. In the debug dropdown, select **Dev Tunnels** > **Create a Tunnel**.
+
+   :::image type="content" source="../assets/images/Copilot/bot-based-VS-dev-tunnel.png" alt-text="Screenshot shows the create a tunnel option in Visual Studio.":::
+
+   1. Select the Account to use to create the tunnel. Azure, Microsoft Account (MSA), and GitHub are the account types that are supported.
+   1. Name: Enter a Name for the tunnel.
+   1. Tunnel Type: Select Persistent or Temporary.
+   1. Access: Select Public.
+   1. Select **OK**. Visual Studio displays a confirmation message that a tunnel is created.
+
+    The tunnel you've created is listed under **Dev Tunnels(MyPublicDevTunnel)**.
+
+1. Go to **Solution Explorer**.
+1. Select your project and right-click.
+1. From the menu, select **Teams Toolkit** > **Prepare Teams App Dependencies**.
+
+   :::image type="content" source="../assets/images/Copilot/bot-based-VS-teams-app-dependencies.png" alt-text="Screenshot shows the Prepare Teams app dependencies option under Teams Toolkit in Visual Studio app project.":::
+
+   1. If prompted, sign in with a Microsoft 365 account.
+1. In the debug dropdown, select **Microsoft Teams (browser)**. Visual Studio launches Teams web client.
+1. Select **Add**.
+1. You can search NuGet package from compose message area, or from the command box.
+
+To trigger the message extension through Copilot, follow these steps:
+
+1. In the debug dropdown, select **Copilot (browser)**. Visual Studio launches Teams web client.
+
+   :::image type="content" source="../assets/images/Copilot/bot-based-VSC-debug-Copilot.png" alt-text="Screenshot shows the Copilot (Browser)  debug option in Visual Studio.":::
+
+1. Go to **Teams web client** > **Apps**.
+1. Search for Copilot and open the Copilot app.
```

Comment:
Latest branding guidance is to continue using "Microsoft 365 Chat" (or *M365 Chat* as it currently appears in the UI) for 11/15

Reply:
Updated.